src/Domain/Communicator/WatchdogCommunicator.php

```diff
@@ -116,7 +117,8 @@ private function getSessionAPIBaseUrl(): string
     {
         $sessionId = $this->gameList->getId();
         if (getenv('DOCKER') !== false) {
-            return 'http://caddy:80/'.$sessionId.'/';
+            return 'http://'.($_ENV['WEB_SERVER_HOST'] ?? 'caddy').':'.($_ENV['WEB_SERVER_PORT'] ?? 80).'/'.$sessionId.
+                '/';
```

Comment:
please sift this out for a separate merge into dev so it can become part of 4.0.2